implementation.d.ts

```diff
@@ -0,0 +1,18 @@
+interface PromiseResolution<T> {
+	status: 'fulfilled';
+	value: T;
+}
+
+interface PromiseRejection<E> {
+	status: 'rejected';
+	reason: E;
+}
+
+type PromiseResult<T, E> = PromiseResolution<T> | PromiseRejection<E>;
+type PromiseTuple<T extends [any, ...any[]]> = {[P in keyof T]: Promise<T[P]>};
```

Comment:
hmm, does this need `any` here, or can it use `unknown`?

Reply:
Since the library is not implemented in TypeScript, it makes no difference here IMHO. The client code specifies the actual types using type variables/inference.